pkg/operator/ceph/config/store.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config allows a ceph config file to be stored in Kubernetes and mounted as volumes into
+// Ceph daemon containers.
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rook/rook/pkg/clusterd"
+	cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	storeName = "rook-ceph-config"
+
+	configVolumeName = "rook-ceph-config"
+
+	confFileName = "ceph.conf"
+)
+
+// Store manages storage of the Ceph config file shared by all daemons (if applicable) as well as an
+// updated 'mon_host' which can be mapped to daemon containers and referenced in daemon command line
+// arguments.
+type Store struct {
+	configMapStore *k8sutil.ConfigMapKVStore
+	context        *clusterd.Context
+}
+
+// GetStore returns the Store for the cluster.
+func GetStore(context *clusterd.Context, namespace string, ownerRef *metav1.OwnerReference) *Store {
+	return &Store{
+		configMapStore: k8sutil.NewConfigMapKVStore(namespace, context.Clientset, *ownerRef),
+		context:        context,
+	}
+}
+
+// CreateOrUpdate creates or updates the stored Ceph config based on the cluster info.
+func (s *Store) CreateOrUpdate(clusterInfo *cephconfig.ClusterInfo) error {
+	c := DefaultCentralizedConfigs(s.context.LogLevel)
+
+	/* TODO: in the future merge config overrides from the cluster CRD */
```

Comment:
Do we need the TODO here? I'd suggest tracking with github issues where possible.

Reply:
Hmm. I will add a feature issue for this. I would prefer to revise my TODO instead of eliminate it entirely, as it serves as a placeholder of sorts. For me it's more of a where-does-this go kind of thing. That said, I don't want to add unnecessary clutter that others might find undesirable or confusing, so if you (or others) still prefer this removed I'm not opposed.

Reply:
If it's going to be done with another PR soon, sounds good to leave the TODO there. It's really the long term issues (or that you aren't planning on fixing immediately in another PR) I would want to make sure are tracked with issues.